types/block.go

```diff
@@ -1095,7 +1102,14 @@ func (data *Data) ComputeShares() (NamespacedShares, int) {
 
 	// find the number of shares needed to create a square that has a power of
 	// two width
-	wantLen := paddedLen(curLen)
+	wantLen := int(squareSize * squareSize)
+	if squareSize == 0 {
+		wantLen = paddedLen(curLen)
+	}
+
+	if wantLen < curLen {
+		return nil, 0, errors.New("square size too small to fit block data")
```

Comment:
yes! it can totally occur if a block producer generates the data hash incorrectly

This is checked for during `ProcessProposal` https://github.com/celestiaorg/celestia-app/blob/e18d8d2301a96702e1bf684735a3620eb059b12f/app/process_proposal.go#L88-L94